### PR TITLE
Guess test framework when symfony/phpunit-bridge is used

### PR DESCRIPTION
**Pull request by siad007**
This PR:

- [x] Adds new feature Support of `symfony/phpunit-bridge` and `symfony/flex` projects.
- [x] Covered by tests
- [X] Doc PR: https://github.com/infection/site/pull/XXX

Fixes https://github.com/infection/infection/issues/493
Fixes https://github.com/infection/infection/issues/588

* find test framework for `simple-phpunit` from `symfony/phpunit-bridge` 
* added `bin/` to path, which is related to symfony flex.

### CODE REVIEW

File: src/Finder/TestFrameworkFinder.php [hunk 1]
```diff
@@ -144,15 +145,23 @@ private function findTestFramework(): string
...
+            $candidates[] = 'simple-phpunit.phar';
+        }
+
         $finder = new ExecutableFinder();
 
+        $cwd = getcwd();
+
         foreach ($candidates as $name) {
-            if ($path = $finder->find($name, null, [getcwd()])) {
+            if ($path = $finder->find($name, null, [$cwd, $cwd . '/bin'])) {
```
Comment (borNfreee) [inline src/Finder/TestFrameworkFinder.php]: `[$cwd, $cwd . '/bin']` can be extracted and reused on lines 159, 164

**Comment by siad007** [comment]
@borNfreee So, I added the tests you suggested but i have no clue how to resolve the failure of the mutation coverage with the generated test files. Can you give me a hint to increase the score?

File: tests/Fixtures/e2e/SymfonyFlex/composer.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    "require-dev": {
+        "symfony-flex-demo/flex": "dev-master"
```
Comment (BackEndTea) [inline tests/Fixtures/e2e/SymfonyFlex/composer.json]: What is the reason for this package? Couldn't we just go with requiring symfony/flex ?

Comment (siad007) [reply tests/Fixtures/e2e/SymfonyFlex/composer.json]: There is no reason - just an accident

**Comment by borNfreee** [comment]
> Can you give me a hint to increase the score?

please ignore this issue for now. It complains that MSI is lower that 90% for changed files, but this is what we had before your PR for those files.

There is a [real error](https://ci.appveyor.com/project/borNfreee/infection/builds/21379562/job/y1v6gwfx7j3y45yn#L377) on AppVeyor (for Windows):

```
[Infection\Finder\Exception\FinderException]                                                            
  Unable to locate a phpunit executable on local system. Ensure that phpunit is installed and available.  
```

File: tests/Fixtures/e2e/SymfonyFlex/composer.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+    "require-dev": {
+        "symfony/flex": "^1.1",
```
Comment (borNfreee) [inline tests/Fixtures/e2e/SymfonyFlex/composer.json]: seems like this package does not add any PHPUnit (only in require-dev for testing itself:

https://github.com/symfony/flex/blob/48438c2ef0e9b8d19e861ede35f0118837e6df5c/composer.json#L16-L19

And if this is the case, this test is the same as `tests/Fixtures/e2e/SymfonyBridge`. Shouldn't we test `symfony-website-skeleton` instead?

I've just created project with `website-skeleton`:

```bash
composer show | grep phpunit
symfony/phpunit-bridge              v4.0.4             Symfony PHPUnit Bridge
```

```bash
ls bin
console	phpunit
```

**Comment by borNfreee** [comment]
Thank you @siad007, great first contribution to Infection!

**Pull request merged**